settings.php

```diff
@@ -26,14 +26,136 @@
+        get_string('settings:aws:region', 'local_smartmedia'),
+        get_string('settings:aws:region_help', 'local_smartmedia'),
+        'ap-southeast-2',
+        $regionoptions));
+
+    $settings->add(new admin_setting_configtext('local_smartmedia/sqs_queue_url',
+        get_string('settings:aws:sqs_queue_url', 'local_smartmedia'),
+        get_string('settings:aws:sqs_queue_url_help', 'local_smartmedia'),
+        '',
+        PARAM_URL));
+
+    // Output settings.
+    $settings->add(new admin_setting_heading('outputheader',
+        get_string('settings:output:header', 'local_smartmedia'),
+        get_string('settings:output:header_desc', 'local_smartmedia')));
+
+    $settings->add(new admin_setting_configcheckbox('local_smartmedia/quality_low',
+        get_string('settings:output:quality_low', 'local_smartmedia'),
+        get_string('settings:output:quality_low_help', 'local_smartmedia'), 0));
+
+    $settings->add(new admin_setting_configcheckbox('local_smartmedia/quality_medium',
+        get_string('settings:output:quality_medium', 'local_smartmedia'),
+        get_string('settings:output:quality_medium_help', 'local_smartmedia'), 0));
+
+    $settings->add(new admin_setting_configcheckbox('local_smartmedia/quality_high',
+        get_string('settings:output:quality_high', 'local_smartmedia'),
+        get_string('settings:output:quality_high_help', 'local_smartmedia'), 0));
+
+    $settings->add(new admin_setting_configcheckbox('local_smartmedia/audio_output',
+        get_string('settings:output:audio_output', 'local_smartmedia'),
+        get_string('settings:output:audio_output_help', 'local_smartmedia'), 0));
+
+    $settings->add(new admin_setting_configcheckbox('local_smartmedia/download_files',
+        get_string('settings:output:download_files', 'local_smartmedia'),
+        get_string('settings:output:download_files_help', 'local_smartmedia'), 0));
+
+    // Processing settings.
+    $settings->add(new admin_setting_heading('processingheader',
+        get_string('settings:processing:header', 'local_smartmedia'),
+        get_string('settings:processing:header_desc', 'local_smartmedia')));
+    $settings->add(new admin_setting_configcheckbox('local_smartmedia/proactiveconversion',
+        get_string('settings:processing:proactiveconversion', 'local_smartmedia'),
+        get_string('settings:processing:proactiveconversion_help', 'local_smartmedia'), 1));
+
+    $settings->add(new admin_setting_configcheckbox('local_smartmedia/viewconversion',
+        get_string('settings:processing:viewconversion', 'local_smartmedia'),
+        get_string('settings:processing:viewconversion_help', 'local_smartmedia'), 1));
+
+    $settings->add(new admin_setting_configduration('local_smartmedia/convertfrom',
+        get_string('settings:processing:convertfrom', 'local_smartmedia'),
+        get_string('settings:processing:convertfrom_help', 'local_smartmedia'), '', 7 * DAYSECS));
```

Comment:
This default isn't working correctly. Also ensure the default is similar to the one from the index date('U', strtotime(date('Y-01-01')));